File: taskcenter/src/main/java/com/iwdael/taskcenter/util/Logger.java
package com.iwdael.taskcenter.util;

/**
 * @author : iwdael
 * @mail : iwdael@outlook.com
 * @project : https://github.com/iwdael/TaskCenter
 */
public class Logger {
    public static Log tag(String tag) {
        return new Log(tag);
    }

    public static class Log {
        private final String tag;

        public Log(String tag) {
            this.tag = tag;
        }

        public void v(String content) {
            android.util.Log.v("task-" + tag, content);
        }

        public void w(String content) {
            android.util.Log.w("task-" + tag, content);
        }
    }
}

File: taskcenter/src/main/java/com/iwdael/taskcenter/util/Logger.kt
package com.iwdael.taskcenter.util

class Logger {
    companion object {
        const val TAG = "task-center"
        fun tag(tag: String) = Log(tag)
    }

    class Log(private val tag: String) {
        fun v(format: String, vararg args: Any) {
            println("$TAG-${tag}: ${java.lang.String.format(format, args)}")
        }

        fun e(format: String, vararg args: Any) {
            System.err.println("$TAG-${tag}: ${java.lang.String.format(format, args)}")
        }
    }
}